tesseract_common/include/tesseract_common/clone_cache.h

```diff
@@ -165,17 +194,33 @@ class CloneCache
 protected:
   void createClone()
   {
-    std::shared_ptr<CacheType> clone;
     if (original_ == nullptr)
       return;
 
-    clone = original_->clone();
+    std::shared_ptr<CacheType> clone = getClone();
+    if (clone == nullptr)
+      return;
 
     // Lock cache
     std::unique_lock<std::mutex> lock(cache_mutex_);
     cache_.push_back(clone);
   }
 
+  std::shared_ptr<CacheType> getClone() const
+  {
+    std::shared_ptr<CacheType> clone;
+    try
+    {
+      clone = original_->clone();
+    }
+    catch (std::exception& e)
+    {
+      CONSOLE_BRIDGE_logError("Clone Cache failed to update cache with the following exception: %s", e.what());
+      return nullptr;
+    }
+    return clone;
```

Comment:
I do not think so and if it did I would believe there is a bug somewhere that needs to be tracked down. I noticed that the original implementation was doing this in one place but not all when I wrote a unit test to test the throw an exception in clone. I assume this could be possible depending on what class is passed to it, but I believe this would be an unlikely case because if the original was good then the clone should not have an issues unless there is a bug I assume. What do you think we should do here? We could remove it  and let the class that uses this decide to catch if this is occurring, because I believe this would mean there is an issue somewhere in the class being passed.